src/download_helper.cc

```diff
@@ -237,23 +237,45 @@ std::shared_ptr<RequestGroup>
 createBtMagnetRequestGroup(const std::string& magnetLink,
                            const std::shared_ptr<Option>& optionTemplate)
 {
-  auto option = util::copy(optionTemplate);
-  auto gid = getGID(option);
-  auto rg = std::make_shared<RequestGroup>(gid, option);
   auto dctx = std::make_shared<DownloadContext>(METADATA_PIECE_SIZE, 0);
 
   // We only know info hash. Total Length is unknown at this moment.
   dctx->markTotalLengthIsUnknown();
-  rg->setFileAllocationEnabled(false);
-  rg->setPreLocalFileCheckEnabled(false);
+
   bittorrent::loadMagnet(magnetLink, dctx);
   auto torrentAttrs = bittorrent::getTorrentAttrs(dctx);
+
+  if (optionTemplate->getAsBool(PREF_BT_LOAD_SAVED_METADATA)) {
+    // Try to read .torrent file saved by aria2 (see
+    // UTMetadataPostDownloadHandler and --bt-save-metadata option).
+    auto torrentFilename =
+        util::applyDir(optionTemplate->get(PREF_DIR),
+                       util::toHex(torrentAttrs->infoHash) + ".torrent");
+
+    bittorrent::ValueBaseBencodeParser parser;
+    auto torrent = parseFile(parser, torrentFilename);
+    if (torrent) {
```

Comment:
Shouldn't we also check that infohash in file is the same as requested in magnet link? This would protect user from accidentally overwriting torrent with another torrent, for example.